chainer/utils/sparse.py

```diff
@@ -53,22 +63,34 @@ def __init__(self, data, row, col, shape, order=None,
 
     def to_dense(self):
         """Returns a dense matrix format of this sparse matrix."""
-        data = self.data.data
-        xp = backend.get_array_module(data)
+        data = self.data
         if data.ndim == 1:
-            x = xp.zeros(self.shape, dtype=data.dtype)
-            nnz = xp.count_nonzero(data)
-            x[self.row[:nnz], self.col[:nnz]] = data[:nnz]
-            return x
-        if data.ndim == 2:
-            nb = data.shape[0]
-            x = xp.zeros((nb, self.shape[0], self.shape[1]),
-                         dtype=data.dtype)
-            for i in range(nb):
-                nnz = xp.count_nonzero(data[i])
-                x[i, self.row[i, :nnz],
-                  self.col[i, :nnz]] = data[i, :nnz]
```

Comment:
The above two lines were the cause of the issue for duplicate indices not taken into account.